bilge-impl/src/bitsize_internal.rs

```diff
@@ -67,11 +92,12 @@ fn generate_struct(struct_data: &ItemStruct, arb_int: &TokenStream) -> TokenStre
     let const_ = if cfg!(feature = "nightly") { quote!(const) } else { quote!() };
 
     quote! {
-        #vis struct #ident {
+        #vis struct #ident #generics #where_clause {
             /// WARNING: modifying this value directly can break invariants
             value: #arb_int,
+            _phantom: ::core::marker::PhantomData<(#(#phantom),*)>
```

Comment:
is it correct that adding a `PhantomData` here is only needed when generics are involved?

if so, do you think it's possible (and practical) to _not_ generate a `_phantom` field unless it's actually needed, just by looking at `item.generics`?

Reply:
It should be possible, yes, but it would create additional conditions at the construction sites (there are 4 in this PR). We could refactor those to use a common constructor method, but I figured I'd go for the approach that requires the least conditional code.

I'm fine either way, though.